src/Particle/ParticleSet.cpp

```diff
@@ -448,24 +446,10 @@ bool ParticleSet::makeMoveAndCheck(Index_t iat, const SingleParticlePos_t& displ
   return is_valid;
 }
 
-bool ParticleSet::makeMoveAndCheck(Index_t iat, const SingleParticlePos_t& displ, const RealType& sdispl)
+bool ParticleSet::makeMoveAndCheckWithSpin(Index_t iat, const SingleParticlePos_t& displ, const RealType& sdispl)
 {
-  activePtcl = iat;
-  activePos  = R[iat] + displ;
-  activeSpin = spins[iat] + sdispl;
-  bool is_valid = true;
-  if (Lattice.explicitly_defined)
-  {
-    if (Lattice.outOfBound(Lattice.toUnit(displ)))
-      is_valid = false;
-    else
-    {
-      newRedPos = Lattice.toUnit(activePos);
-      if (!Lattice.isValid(newRedPos)) is_valid = false;
-    }
-  }
-  computeNewPosDistTablesAndSK(iat, activePos);
-  return is_valid;
+    makeMoveAndCheck(iat,displ);
+    activeSpin = spins[iat]+sdispl;
```

Comment:
The check result needs to be propagated.
```
activeSpin = spins[iat]+sdispl;
return makeMoveAndCheck(iat,displ);
```

Reply:
oops, fixed